crates/router/src/connector/utils.rs

```diff
@@ -414,6 +422,27 @@ impl AddressDetailsData for api::AddressDetails {
     }
 }
 
+pub trait MandateData {
+    fn get_end_date(&self, format: date_time::DateFormat) -> Result<String, Error>;
+    fn get_meta_data(&self) -> Result<pii::SecretSerdeValue, Error>;
+}
+
+impl MandateData for payments::MandateAmountData {
```

Comment:
```suggestion
impl payments::MandateAmountData {
```

Reply:
MandateData trait is not required.

Reply:
<img width="759" alt="Screen Shot 2023-04-10 at 12 55 02 PM" src="https://user-images.githubusercontent.com/20727598/230849954-f4f51b27-c9cf-4621-bfaa-88dcf8e51809.png">
getting this error